homeassistant/components/weather/yandex.py

```diff
@@ -0,0 +1,218 @@
+"""
+Support for Yandex Weather.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/weather.yandex/
+"""
+import logging
+from datetime import datetime, timedelta
+from contextlib import wraps
+
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.components.weather import (
+    WeatherEntity, PLATFORM_SCHEMA, ATTR_FORECAST_CONDITION,
+    ATTR_FORECAST_PRECIPITATION, ATTR_FORECAST_TEMP,
+    ATTR_FORECAST_TEMP_LOW, ATTR_FORECAST_TIME)
+from homeassistant.const import \
+    CONF_NAME, TEMP_CELSIUS, CONF_LATITUDE, CONF_LONGITUDE, CONF_API_KEY
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers import config_validation as cv
+
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['yandex-weather-api==0.2.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = 'Яндекс.Погода'
+
+# API limit is 50 req/day
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=40)
+
+CONDITION_TO_HA = {
+    "clear": "clear-night",
+    "partly-cloudy": "partlycloudy",
+    "cloudy": "cloudy",
+    "overcast": "cloudy",
+    "partly-cloudy-and-light-rain": "rainy",
+    "partly-cloudy-and-rain": "pouring",
+    "overcast-and-rain": "rainy",
+    "overcast-thunderstorms-with-rain": "lightning-rainy",
+    "cloudy-and-light-rain": "rainy",
+    "overcast-and-light-rain": "rainy",
+    "cloudy-and-rain": "rainy",
+    "overcast-and-wet-snow": "snowy-rainy",
+    "partly-cloudy-and-light-snow": "snowy",
+    "partly-cloudy-and-snow": "snowy",
+    "overcast-and-snow": "snowy",
+    "cloudy-and-light-snow": "snowy",
+    "overcast-and-light-snow": "snowy",
+    "cloudy-and-snow": "snowy"
+}
```

Comment:
The Yandex conditions must be mapped to the [Recommended values for state and condition
](https://developers.home-assistant.io/docs/en/entity_weather.html#recommended-values-for-state-and-condition)

Reply:
I suggest that you do a dict lookup like the other platforms do. This way the platform is easier to maintain if we add more conditions.

Reply:
I'm sorry, I guess my english is quite bad - I can't understand what suggestions are you proposing. I am doing a dict lookup [in `condition()` property](https://github.com/ASMfreaK/home-assistant/blob/yandex-weather/homeassistant/components/weather/yandex.py#L162) just like other platforms.

Reply:
Do the mapping like the other platforms, e.g.,

```python
CONDITION_CLASSES = {
   ....
    "rainy": ["cloudy-and-light-rain", "overcast-and-light-rain" ],
    "snowy": ["cloudy-and-light-snow", "overcast-and-light-snow", "cloudy-and-snow"],
}